Fixed an issue when using a saved card at checkout

fiddledsticks opened the pull request:
There was an issue connecting to the Stripe gateway when using a saved card. After a bit of probing I found out it was because the card field been sent was blank. I solved this by changing the field that is been checked to `chosen_card` instead of `save_card`. It all seems to be working as expected now charges go through for new card (not saved), new cards (saved), and saved cards.

maximetalhouk commented:
Hello,

I tried to apply this modification, but unfortunately, I still can't use saved cards in test mode. Is this related to the use of https protocol?

Thank you for your help.

fiddledsticks commented:
I was working on this earlier today, and have come across an issue with this solution, I'll pull in another, better solution soon. Your issue could be related to HTTPS though, as that is necessary for Stripe.

fiddledsticks commented:
There is another bug that I'll send another PR for if I find a solution where the save_card value can't be changed once you submit the form. E.g. You untick the save card box at checkout, but your card comes back with an error, you fix the error then decide you want to save the card, you tick the box, but the save_card value still goes through as `false`.

stezu commented:
Does this fix the empty card errors people were reporting in the WordPress support threads?

fiddledsticks commented:
That was fixed by #66, so cards are now saving correctly. This fixes an issue where you couldn't use a saved card at checkout, it was sending the wrong field, so I updated the field name and added some additional logic to detect if a card was being saved or if it was a saved card.
